internal/minecraft/client.go

```diff
@@ -46,28 +46,53 @@ const (
 	ClientStatePlayerLogin
 )
 
+// Handshake is the first packet sent to a Minecraft server.
+//
+// See: https://wiki.vg/Protocol#Handshaking
```

Comment:
It was time to strongly type the handshake type, so I added that here and changed the code as required to handle it.